src/par_iter/collect/mod.rs

```diff
@@ -0,0 +1,38 @@
+use super::ExactParallelIterator;
+use std::isize;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+mod consumer;
+use self::consumer::CollectConsumer;
+
+pub fn collect_into<PAR_ITER,T>(mut pi: PAR_ITER, v: &mut Vec<T>)
+    where PAR_ITER: ExactParallelIterator<Item=T>,
+          PAR_ITER: ExactParallelIterator,
+          T: Send,
+{
+    let len = pi.len();
+    assert!(len < (isize::MAX) as usize);
+
+    v.truncate(0); // clear any old data
+    v.reserve(len); // reserve enough space
+    let target = v.as_mut_ptr(); // get a raw ptr
+    let writes = AtomicUsize::new(0);
+    let consumer = unsafe { // assert that target..target+len is unique and writable
+        CollectConsumer::new(&writes, target, len)
+    };
+    pi.drive(consumer);
```

Comment:
Along those lines, I've noticed I don't have many actual use cases for `collect`. It *seems* handy, but all the tests I've written tend to use other things, or just use `for_each` and write directly into the output buffer (as you suggested). Basically the only advantage `collect` brings (right now) is handling uninitialized memory for you; I guess if it supported `filter` it would be mildly more interesting.